### ISSUE TITLE
please explain MIDI routing

### ISSUE THREAD

**Issue by jwaldmann** [spell-music/csound-expression#51]
The documentation (Patches.md) says ".. way we can play the patches with real MIDI-device:"
```
dac $ mul 0.75 $ atMidi toneWheelOrgan
```
but on my machine this fails with
```
ALSA midi: Using all devices.
ALSA lib rawmidi_hw.c:233:(snd_rawmidi_hw_open) open /dev/snd/midiC2D0 failed: Device or resource busy
ALSA: error opening MIDI input device: 'hw:2,0'
ALSA midi: Error opening device: hw:2,0
ALSA midi: No devices found.
```
How is this supposed to work? I was expecting that some midi port shows up (e.g., in `aplaymidi -l`) so that I can route signals there from `qjackctl`.

**Comment by anton-k** [comment]
I think that we need to research on how to setup the Csound flags

https://csound.com/docs/manual/CommandFlagsCategory.html

See the section: MIDI Realtime Input/Ouput

I'm myself trying various options.. till I find something that works
Sometimes it helps to stop ALSA Midi  bridge in the cadence

**Comment by anton-k** [comment]
Maybe you need to add `-+rtmidi=alsa` if you are referring to device by hw:CARD,DEVICE

```
def { rtmidi = Just AlsaMidi } <> setMidiDevice "hw:2,0,0"
```

**Comment by anton-k** [comment]
I think this is not the issue of CE. It's issue of Linux+midi workflow

**Comment by jwaldmann** [comment]
Yes of course. Still, perhaps some other CE user knows how to handle this. But feel free to close the issue.
